### PR TITLE
Upgrade to Infinispan version 12

### PR DESCRIPTION
**Pull request by Scuilion**

### CODE REVIEW

**Comment by karesti** [comment]
@Scuilion sorry I did not see this pr and I moved master to main! thanks for the upgrade, if you reopen a new pull request I will be more than happy to merge

File: README.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
 # Run infinispan server
 
-```bash 
-docker run -it -p 11222:11222 jboss/infinispan-server:9.4.0.Final
+```bash
+docker run -p 11222:11222 -e USER="user1" -e PASS="user1pass" infinispan/server:12.1.7.Final-1
```
Comment (karesti) [inline README.md:4]: This should be admin/password since admin has admin role by default since 12

Comment (karesti) [reply README.md:4]: the name of the image should not be Final-1 ...

File: reader/reader-near-code/src/main/resources/hotrod-client.properties [hunk 2]
```diff
@@ -1,4 +1,11 @@
 infinispan.client.hotrod.server_list=127.0.0.1:11222
 # Allows to connect to docker container server from Mac:
 # Read https://blog.infinispan.org/2018/03/accessing-infinispan-inside-docker-for.html
-infinispan.client.hotrod.client_intelligence=BASIC
\ No newline at end of file
+infinispan.client.hotrod.client_intelligence=BASIC
+infinispan.client.hotrod.marshaller=org.infinispan.commons.marshall.JavaSerializationMarshaller
```
Comment (karesti) [inline reader/reader-near-code/src/main/resources/hotrod-client.properties:5]: I would like to move this to Protobuf, I will deal this after this PR

File: writer/src/main/java/org/infinispan/writer/ContributorsDataLoader.java [hunk 3]
```diff
@@ -25,7 +25,7 @@ public void run(String... args) throws Exception {
       logger.info("            LOAD CONTRIBUTORS            ");
       logger.info("=========================================");
       RemoteCache<Integer, Contributor> contributors = cacheManager.administration()
```
Comment (karesti) [inline writer/src/main/java/org/infinispan/writer/ContributorsDataLoader.java:27]: we can do this better now. Look to the simple tutorials https://github.com/infinispan/infinispan-simple-tutorials/tree/main/integrations/spring-boot/cache-remote

**Review by karesti — COMMENTED** [review]
Happy to see this contribution!! please grab a look to my review and sorry again for renaming master before realising this contribution!

File: writer/src/main/java/org/infinispan/writer/ContributorsDataLoader.java (hunk 3 above)
Comment (karesti) [reply writer/src/main/java/org/infinispan/writer/ContributorsDataLoader.java:27]: here https://github.com/infinispan/infinispan-simple-tutorials/blob/main/integrations/spring-boot/cache-remote/src/main/java/org/infinispan/tutorial/simple/spring/remote/InfinispanConfiguration.java#L19

**Comment by Scuilion** [comment]
Awesome, thanks for the feedback. Will get another PR out shortly.

**Pull request closed without merging**